delta/extensions/losses.py

```diff
@@ -199,7 +199,7 @@ def preprocess(self, y_true, y_pred):
             nodata = tf.expand_dims(nodata, -1)
 
         # zero all nodata entries
-        y_pred = y_pred * tf.cast(tf.logical_not(nodata), tf.float32)
+        y_pred = tf.cast(y_pred, tf.float32) * tf.cast(tf.logical_not(nodata), tf.float32)
```

Comment:
This isn't immediately related to the mixed precision change. I was getting an error when trying to run with mixed precision because you can't hardcode variable types since mixed precision will dynamically change variable types depending on the variable's location in the model, need for accuracy, etc. Is this casting crucial? A test spinal model was training fine with this change.

Reply:
I think this change should be fine. Will let you know if I see any problems.